devimint/src/federation.rs

```diff
@@ -242,6 +242,23 @@ impl Federation {
         Ok(())
     }
 
+    pub async fn pegin_client(&self, amount: u64, client: &Client) -> Result<()> {
+        info!(amount, "Pegging-in client funds");
+        let deposit = cmd!(client, "deposit-address").out_json().await?;
+        let deposit_address = deposit["address"].as_str().unwrap();
+        let deposit_operation_id = deposit["operation_id"].as_str().unwrap();
+
+        self.bitcoind
+            .send_to(deposit_address.to_owned(), amount)
+            .await?;
+        self.bitcoind.mine_blocks(21).await?;
+
+        cmd!(client, "await-deposit", deposit_operation_id)
+            .run()
+            .await?;
+        Ok(())
+    }
```

Comment:
The `pegin` function should be deduplicate with this one.

Reply:
Made `pegin` delegate to `pegin_client`. I should fully remove pegin in a rewrite of #3698 which is now quite stale